packages/@vuepress/shared/src/types/page.ts

```diff
@@ -51,6 +51,11 @@ export type PageData<
    * Headers of the page
    */
   headers: PageHeader[]
+
+  /**
+   * Route meta data of the page
+   */
+  meta?: Record<string, unknown>
```

Comment:
https://github.com/vuepress/vuepress-next/blob/3746e664d79dc1088e94bbb6f2eda2cf6df7cc1c/packages/%40vuepress/core/src/types/page.ts#L7

Then `meta` would also be part of `Page`, so it should be added to the docs: [Node API > Page properties]